<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
<ol style="list-style-type: lower-alpha">
<li>Introduction to shroom</li>
</ol> • shroom</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="&lt;ol style=" list-style-type: lower-alpha>
</head>
<body>
<li>Introduction to shroom</li>
" /&gt;

<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">shroom</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1-introduction.html">a) Introduction to the shroom package</a>
    </li>
    <li>
      <a href="../articles/a2-intro_to_ind_project_questions.html">a) Introduction to shroom</a>
    </li>
    <li>
      <a href="../articles/b-loading_the_shroom_package.html">b) Loading the shroom package</a>
    </li>
    <li>
      <a href="../articles/c-subsetting_a_dataframe_with_dplyr.html">c) Subsetting a dataframe with dplyr</a>
    </li>
    <li>
      <a href="../articles/d-summarizing_data_with_dplyr.html">d) Summarizing data with dplyr</a>
    </li>
    <li>
      <a href="../articles/f1-analysis_walkthrough_t_test_and_friends.html">f) Analysis walthrough: t-test and friends</a>
    </li>
    <li>
      <a href="../articles/f3-analysis_walkthrough_2-way_ANOVA.html">f) Analysis walthrough: 2-way factorial model</a>
    </li>
    <li>
      <a href="../articles/f4-analysis_walkthrough_nested_ANOVA.html">f) Analysis walthrough: 2-way factorial model</a>
    </li>
    <li>
      <a href="../articles/g1-calculating_effect_sizes_with_metafor.html">e) Calculating effect sizes using the metafor package</a>
    </li>
    <li>
      <a href="../articles/g2-analysis_of_difference_scores.html">f) Analysis of difference scores: thinking about effect sizes, sample sizes &amp; effect size variances</a>
    </li>
    <li>
      <a href="../articles/s1-stub-probablistic_index_models.html">s1) Stub - Probablistic index models</a>
    </li>
    <li>
      <a href="../articles/y-creating_a_package.html">w) Creating this package</a>
    </li>
    <li>
      <a href="../articles/z-build_and_rebuild_package.html">z) Building and rebuilding vignettes</a>
    </li>
    <li>
      <a href="../articles/z2-building_a_package.html">z2) Building and rebuilding vignettes - orig</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1><ol style="list-style-type: lower-alpha">
<li>Introduction to shroom</li>
</ol></h1>
                        <h4 class="author">Nathan Brouwer</h4>
            
            <h4 class="date">2018-11-28</h4>
      
      
      <div class="hidden name"><code>a2-intro_to_ind_project_questions.Rmd</code></div>

    </div>

    
    
<div id="the-deep-shroom-analysis-issues" class="section level2">
<h2 class="hasAnchor">
<a href="#the-deep-shroom-analysis-issues" class="anchor"></a>The deep shroom analysis issues</h2>
<p>[not yet edited]</p>
<p>After having spent way too much time navel-gazing about the fly data and how to make best use of it I’ve decided to propose 2 complementary types of questions for students to address. First, I have a small set of biological questions to chose from. Second, I have a number of statistical questions related to how this type of data set can be analyzed. Each student will pick one of the biological questions (or potentially propose their own) and one of the statistical questions, then decide whether to address the question using all of the data or a subset of the data, with this decision potentially impacting how the data are worked with. Students will then set up the data as needed using dplyr functions like filter(), carry out the analyses, and write a paper discussing the biological results and if the statistical issues they looked in to impact those results.</p>
<p>Below is an outline of the biological and statistical questions. After that are more details.</p>
</div>
<div id="outline" class="section level2">
<h2 class="hasAnchor">
<a href="#outline" class="anchor"></a>Outline</h2>
<div id="biological-questions" class="section level3">
<h3 class="hasAnchor">
<a href="#biological-questions" class="anchor"></a>Biological questions:</h3>
<p>Students will pick one question, or develop their own. Different biological foci might impact how the data are analyse and the types of statistical questions that can be investigated or the subset of the data used.</p>
<ul>
<li>What is the impact of maleness?</li>
<li>What is the impact of each of the 18 mutant candidate alleles?</li>
<li>What is the impact of the 6 candidate genes?</li>
</ul>
</div>
<div id="statistical-questions" class="section level3">
<h3 class="hasAnchor">
<a href="#statistical-questions" class="anchor"></a>Statistical questions:</h3>
<ul>
<li>How does “pooling” the data from all of the students impact the results?</li>
<li>How do different approaches to correcting for multiple comparisons impact the results?</li>
<li>Can we use the technique known as “equivalence tests” to determine if any alleles are genes truly have no mean effect? (That is, <em>accept</em> the null hypothesis)</li>
<li>Does the use of different types of <strong>effect sizes</strong> impact the results?</li>
<li>What is the impact of different approaches to <strong>weighting</strong> the results of crosses with different sample sizes?</li>
<li>What is the impact assuming normality and using t-tests versus using non-parametric methods that respect the ordinal nature of the data?</li>
</ul>
</div>
</div>
<div id="how-much-data-to-use" class="section level2">
<h2 class="hasAnchor">
<a href="#how-much-data-to-use" class="anchor"></a>How much data to use?</h2>
<ul>
<li>Data from all 14 lab sections combined</li>
<li>Data from just one lab section</li>
<li>Data for just a single gene</li>
</ul>
</div>
<div id="details" class="section level2">
<h2 class="hasAnchor">
<a href="#details" class="anchor"></a>Details</h2>
<div id="biological-questions-1" class="section level3">
<h3 class="hasAnchor">
<a href="#biological-questions-1" class="anchor"></a>Biological questions:</h3>
<div id="what-is-the-impact-of-maleness" class="section level4">
<h4 class="hasAnchor">
<a href="#what-is-the-impact-of-maleness" class="anchor"></a>What is the impact of maleness?</h4>
<p>In this class we work with flies that have been engineered to overexpress shroom. This is done by inserting an extra copy of shroom on the X-chromosome. Flies therefore have their normal autosomal copy (on chromosome 3 I believe) and an extra copy linked to molecular switch that allows us to turn in on in specific tissues during development, such as the wing or eye.</p>
<p>Male fruit flies are XY and typically have higher levels of Shroom over expression due to the process of <strong>dosage compensation.</strong> Our fly dataset can be used to investigate how dosage compensation varies between strains of flies with different genomic backgrounds. To address this question, you could look at just “control” flies and determine if males always a different from females to the same degree, or if there is variation among the genes or allele.s</p>
<p><strong>References</strong>:</p>
<p>Jaffe and Laird. 1986. Dosage compensation in Drosophila. Trends IN Genetics.</p>
<p>Mank, J. 2009. The W, X, Y and Z of sex-chromosome dosage compensation. Trend in Ecology &amp; Evolution</p>
<p><br></p>
</div>
<div id="what-is-the-impact-of-each-of-the-18-mutant-candidate-alleles" class="section level4">
<h4 class="hasAnchor">
<a href="#what-is-the-impact-of-each-of-the-18-mutant-candidate-alleles" class="anchor"></a>What is the impact of each of the 18 mutant candidate alleles?</h4>
<p>This study involves 6 genes and 3 alleles of each gene. Which allele shows the strongest impact on the fly phenotype? Is there one allele that stands out from all the rest, or are there several alleles tied for the “best” candidate? Are there any alleles for which we should conclude that there is no effects (that is, accept the null?).</p>
</div>
<div id="what-is-the-impact-of-the-6-candidate-genes" class="section level4">
<h4 class="hasAnchor">
<a href="#what-is-the-impact-of-the-6-candidate-genes" class="anchor"></a>What is the impact of the 6 candidate genes?</h4>
<p>Instead of focusing on each seperate allele, we can focus on the 6 genes. By (more or less) averaging the effect of all three alleles we can determine how strong and consistent the effect of a mutation in that gene is. In some cases, two alleles might have a negative effect on a phenotypic score, while the third might have the opposite effect, increasing the score. Should we conclude that there is a negative effect, or does the one positive effect cancel out the two negative ones?</p>
</div>
</div>
<div id="statistical-questions-1" class="section level3">
<h3 class="hasAnchor">
<a href="#statistical-questions-1" class="anchor"></a>Statistical questions:</h3>
<div id="how-does-pooling-the-data-from-all-of-the-students-impact-the-results" class="section level4">
<h4 class="hasAnchor">
<a href="#how-does-pooling-the-data-from-all-of-the-students-impact-the-results" class="anchor"></a>How does “pooling” the data from all of the students impact the results?</h4>
<p>We ask the students in the class to focus mostly on the data they generate themselves. We also ask them to consider the data from the entire class and compare their results to the class average. For the whole-class dataset, we pool all the data together, essentially treating the data as if they were collected by a single student. We are assuming that the biggest source of variation in the data is due to the effect of the mutant alleles, and that variation in how students score the flies isn’t important. Statistically, this is wrong, but how much does it impact our results? <strong>Do any conclusions change between a correct analysis of the data which recognizes the “nested” nature of the data and an naive analysis the acts as a single student scored all the flies?</strong></p>
</div>
<div id="how-do-different-approaches-to-correcting-for-multiple-comparisons-impact-the-results" class="section level4">
<h4 class="hasAnchor">
<a href="#how-do-different-approaches-to-correcting-for-multiple-comparisons-impact-the-results" class="anchor"></a>How do different approaches to correcting for multiple comparisons impact the results?</h4>
<p>We are looking at 6 genes and 18 alleles, with the goal of selecting strong candidates for future analysis. However, we know from thinking about multiple comparisons that we could end up with <strong>false positives</strong> just due to chance. We can correct for this using a number of methods. Traditionally, biologists have used the <strong>Bonferoni</strong> method because it is really simple to do. The Bonferoni approach, however, can impose a stiff penalty and reduce power, and you can end up getting “Bonferonied into oblivion” (TL Ashman, 2009). Some ecologists have been fed up with this said we should not bother with comparisons for multiple comparisons (Moran 2003). Epidemiologists have also voiced opinions like this; I’m sure in all disciplines you’ll find multiple comparisons backlash. In the 1990s some modifications to Bonferoni started to be used, such as the Holm and Hochberg methods. These are bit harder to do by hand (though R can do them very easily). In the last 10 years, the <strong>False Discovery Rate</strong> (FDR) has become very popular in some fields, and some people in biology have said it should be used rather than other methods (Pike 2011). <strong>How do the results of the study vary between different approaches to multiple comparisons? Do we benefit from using the FDR? Do things even change using Bonferonni?</strong></p>
<p><strong>References</strong> Moran, MD. 2003. Arguments for rejecting the sequential Bonferroni in ecological studies. Oikos.</p>
<p>Pike 2011. Using false discovery rates for multiple comparisons in ecology and evolution Methods in E &amp; E.</p>
</div>
<div id="can-we-use-the-technique-known-as-equivalence-tests-to-determine-if-any-alleles-are-genes-truly-have-no-mean-effect-that-is-accept-the-null-hypothesis" class="section level4">
<h4 class="hasAnchor">
<a href="#can-we-use-the-technique-known-as-equivalence-tests-to-determine-if-any-alleles-are-genes-truly-have-no-mean-effect-that-is-accept-the-null-hypothesis" class="anchor"></a>Can we use the technique known as “equivalence tests” to determine if any alleles are genes truly have no mean effect? (That is, <em>accept</em> the null hypothesis)</h4>
<p>The logic of p-values doesn’t allow us to ever accept a null hypothesis, no matter how large the p-value. Recently, the technique of <strong>equvilance testing</strong> is becoming more popular (Motulsky 2018). Its argued that equivalence testing allows us to conclude that the null hypothesis is indeed true. <strong>If we use equivalence testing, can we conclude that any genes or alleles truly have no impact and can be confidently ignored in future studies?</strong></p>
<p><strong>References</strong></p>
<p>Motulsky. Chapter 21: Testing for equivalence or noninferiority.</p>
</div>
<div id="does-the-use-of-different-types-of-effect-sizes-impact-the-results" class="section level4">
<h4 class="hasAnchor">
<a href="#does-the-use-of-different-types-of-effect-sizes-impact-the-results" class="anchor"></a>Does the use of different types of <strong>effect sizes</strong> impact the results?</h4>
<p>The data produced by this study are very complicated. One way of making this simpler is to summarize the data produced by each student and calculate an effect size statistics for the difference between the students experimental and control flies. These effect sizes can then be used as the raw data for further analysis.</p>
<p>For example, if want to analyze the raw data we have to account for the fact that all flies scored by the same student are correlated with each other because each student will have a slightly different internal calibration to the score system. We can therefore say that our data has a hierarchical or nested structure, with each student acting as a node in the hierarchy, and each fly being “nested” within the node of the student which scored it. We have 250 students and about 3000 female flies. We therefore have 3000 raw data points, but these data points are clustered into 250 groups. If we take each student’s data and turn it into effect sizes, we collapse this large dataset down into just 250 data points, one for each student (assuming we are just looking at one sex). This makes our analyses much simpler and cleaner.</p>
<p>Converting the data from each student into an effect size creates a situation that is similar to a <strong>meta-analysis</strong> (Motulsky 2018). Meta-analysis is a way to summarize data from multiple independent studies. It was first used to summarize medical studies and has become popular in the last 25 years in ecology. Recently, meta-analysis has been used to address debates in lab-based biology (Nature 2016; Dent et al 2016).</p>
<p>The simplest effect sizes is just the difference between the means (<strong>mean difference</strong>) of the experimental and control flies from a single student. An alternative that accounts for the variation in each group of flies is an effect size like <strong>Cohen’s D</strong>, which is a <strong>Standardized mean difference</strong> (SMD). Cohen’s D, however, is rooted in assumptions about normality, and our data is definitely not normal! An alternative that might be better is the <strong>log response ratio</strong> (log(mean1/mean2)). <strong>If we take a meta-analytic approach, does the type of effect size impact the results</strong>?</p>
<p><strong>References</strong> Motulsky, 2018. Chapter 43: Meta-analysis.</p>
</div>
<div id="what-is-the-impact-of-different-approaches-to-weighting-the-results-of-crosses-with-different-sample-sizes" class="section level4">
<h4 class="hasAnchor">
<a href="#what-is-the-impact-of-different-approaches-to-weighting-the-results-of-crosses-with-different-sample-sizes" class="anchor"></a>What is the impact of different approaches to <strong>weighting</strong> the results of crosses with different sample sizes?</h4>
<p>The simplest form of meta-analysis calculated an effect size and uses it as raw data for subsequent analysis. This is easy, but ignores differences in the standard deviation, standard errors, and sample size between the data produced by different students; aberrant results from small sample sizes could potentially through off the results. These differences can be accommodated by providing different <strong>weights</strong> to studies depending on how precise they have estimated their means or how large their sample size. Traditionally, weights have been set as 1/SD for each study. Studies with small standard deviations (SD) therefore have larger weights and have more influence over the results. Recently, evidence has been accumulating that using sample sizes for weights is better. <strong>Does the type of weights used impact our results?</strong>.</p>
</div>
<div id="what-is-the-impact-assuming-normality-and-using-t-tests-versus-using-non-parametric-methods-that-respect-the-ordinal-nature-of-the-data" class="section level4">
<h4 class="hasAnchor">
<a href="#what-is-the-impact-assuming-normality-and-using-t-tests-versus-using-non-parametric-methods-that-respect-the-ordinal-nature-of-the-data" class="anchor"></a>What is the impact assuming normality and using t-tests versus using non-parametric methods that respect the ordinal nature of the data?</h4>
<p>These data are inherently non-normal. It is much easier to treat them as non-normal and just use t-tests. Rank-based methods are arguable more important, but does it matter. <strong>Does using t-tests instead of rank-bsed methods impact our results</strong></p>
</div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#the-deep-shroom-analysis-issues">The deep shroom analysis issues</a></li>
      <li><a href="#outline">Outline</a></li>
      <li><a href="#how-much-data-to-use">How much data to use?</a></li>
      <li><a href="#details">Details</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  
</body>
</html>
