<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html>
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1.0">
<title>
<ol start="6" style="list-style-type: lower-alpha">
<li>Analysis of difference scores: thinking about effect sizes, sample sizes &amp; effect size variances</li>
</ol> • shroom</title>
<!-- jquery --><script src="https://code.jquery.com/jquery-3.1.0.min.js" integrity="sha384-nrOSfDHtoPMzJHjVTdCopGqIqeYETSXhZDFyniQ8ZHcVy08QesyHcnOUpMpqnmWq" crossorigin="anonymous"></script><!-- Bootstrap --><link href="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-BVYiiSIFeK1dGmJRAkycuHAHRg32OmUcww7on3RYdg4Va+PmSTsz/K68vbdEjh4u" crossorigin="anonymous">
<script src="https://maxcdn.bootstrapcdn.com/bootstrap/3.3.7/js/bootstrap.min.js" integrity="sha384-Tc5IQib027qvyjSMfHjOMaLkfuWVxZxUPnCJA7l2mCWNIpG9mGCD8wGNIcPD7Txa" crossorigin="anonymous"></script><!-- Font Awesome icons --><link href="https://maxcdn.bootstrapcdn.com/font-awesome/4.6.3/css/font-awesome.min.css" rel="stylesheet" integrity="sha384-T8Gy5hrqNKT+hzMclPo118YTQO6cYprQmhrYwIiQ/3axmI1hQomh7Ud2hPOy8SP1" crossorigin="anonymous">
<!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/1.7.1/clipboard.min.js" integrity="sha384-cV+rhyOuRHc9Ub/91rihWcGmMmCXDeksTtCihMupQHSsi8GIIRDG0ThDc3HGQFJ3" crossorigin="anonymous"></script><!-- sticky kit --><script src="https://cdnjs.cloudflare.com/ajax/libs/sticky-kit/1.1.3/sticky-kit.min.js" integrity="sha256-c4Rlo1ZozqTPE2RLuvbusY3+SU1pQaJC0TjuhygMipw=" crossorigin="anonymous"></script><!-- pkgdown --><link href="../pkgdown.css" rel="stylesheet">
<script src="../pkgdown.js"></script><meta property="og:title" content="&lt;ol start=" style="list-style-type: lower-alpha">
</head>
<body>
<li>Analysis of difference scores: thinking about effect sizes, sample sizes &amp; effect size variances</li>
" /&gt;

<meta property="og:description" content="">
<meta name="twitter:card" content="summary">
<!-- mathjax --><script src="https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--><div class="container template-article">
      <header><div class="navbar navbar-default navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <span class="navbar-brand">
        <a class="navbar-link" href="../index.html">shroom</a>
        <span class="label label-default" data-toggle="tooltip" data-placement="bottom" title="Released package">0.0.0.9000</span>
      </span>
    </div>

    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
<li>
  <a href="../index.html">
    <span class="fa fa-home fa-lg"></span>
     
  </a>
</li>
<li>
  <a href="../reference/index.html">Reference</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" aria-expanded="false">
    Articles
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
<li>
      <a href="../articles/a1-introduction.html">a) Introduction to the shroom package</a>
    </li>
    <li>
      <a href="../articles/a2-intro_to_ind_project_questions.html">a) Introduction to shroom</a>
    </li>
    <li>
      <a href="../articles/b-loading_the_shroom_package.html">b) Loading the shroom package</a>
    </li>
    <li>
      <a href="../articles/c-subsetting_a_dataframe_with_dplyr.html">c) Subsetting a dataframe with dplyr</a>
    </li>
    <li>
      <a href="../articles/d-summarizing_data_with_dplyr.html">d) Summarizing data with dplyr</a>
    </li>
    <li>
      <a href="../articles/f1-analysis_walkthrough_t_test_and_friends.html">f) Analysis walthrough: t-test and friends</a>
    </li>
    <li>
      <a href="../articles/f3-analysis_walkthrough_2-way_ANOVA.html">f) Analysis walthrough: 2-way factorial model</a>
    </li>
    <li>
      <a href="../articles/f4-analysis_walkthrough_nested_ANOVA.html">f) Analysis walthrough: 2-way factorial model</a>
    </li>
    <li>
      <a href="../articles/g1-calculating_effect_sizes_with_metafor.html">e) Calculating effect sizes using the metafor package</a>
    </li>
    <li>
      <a href="../articles/g2-analysis_of_difference_scores.html">f) Analysis of difference scores: thinking about effect sizes, sample sizes &amp; effect size variances</a>
    </li>
    <li>
      <a href="../articles/s1-stub-probablistic_index_models.html">s1) Stub - Probablistic index models</a>
    </li>
    <li>
      <a href="../articles/y-creating_a_package.html">w) Creating this package</a>
    </li>
    <li>
      <a href="../articles/z-build_and_rebuild_package.html">z) Building and rebuilding vignettes</a>
    </li>
    <li>
      <a href="../articles/z2-building_a_package.html">z2) Building and rebuilding vignettes - orig</a>
    </li>
  </ul>
</li>
<li>
  <a href="../news/index.html">Changelog</a>
</li>
      </ul>
<ul class="nav navbar-nav navbar-right"></ul>
</div>
<!--/.nav-collapse -->
  </div>
<!--/.container -->
</div>
<!--/.navbar -->

      
      </header><div class="row">
  <div class="col-md-9 contents">
    <div class="page-header toc-ignore">
      <h1><ol start="6" style="list-style-type: lower-alpha">
<li>Analysis of difference scores: thinking about effect sizes, sample sizes &amp; effect size variances</li>
</ol></h1>
                        <h4 class="author">Nathan Brouwer</h4>
            
            <h4 class="date">2018-11-28</h4>
      
      
      <div class="hidden name"><code>g2-analysis_of_difference_scores.Rmd</code></div>

    </div>

    
    
<div id="introduction" class="section level2">
<h2 class="hasAnchor">
<a href="#introduction" class="anchor"></a>Introduction</h2>
<p>A major feature of the Neural Defects class that impacts data analysis is the fact that each student visually scores fly wing to character how “crinkly” the wings are. Wings are scored using reference photos from 1 (wildtype) to 8 (wing is almost entirely rolled up and looks more like a piece of string than a wing).</p>
<p>Each student does the scoring on their own while looking at all of their flies. They are instructured to quasi-rank the flies from different groups. For example, to look at all of the male flies (controls and experimental flies) and organize them from most crinkly (8-like) to least crinkly. Each student’s data should therefore be internally consisent and represent any difference between the treatments. However, what one student might scores as a 6 another might score as an 8.</p>
<p>The best way to analyze the wing score data is therefore to treat each student as a data point by</p>
<ol style="list-style-type: decimal">
<li>Taking the mean of all of their flies in each group (male controls, female contorls, male experimental, etc), and</li>
<li>subtracting the mean of their experimental flies from the mean of their control flies for each sex (male control - male experimenta, etc).</li>
</ol>
<p>This produces 14 data points for each allele being studied, because there is 1 student studying each allele in each section of the class, and there are 14 sections of the class. Since there are 14 sections and about 18 students per class we have about 252 data points.</p>
<div id="differene-scores-effect-sizes" class="section level3">
<h3 class="hasAnchor">
<a href="#differene-scores-effect-sizes" class="anchor"></a>Differene scores &amp; effect sizes</h3>
<p>We’ll call the differences between treatments for a single student’s <strong>difference scores</strong> to give them a sciencey sounding name. These difference scores act as an unstandardized <strong>effect size</strong> for each student’s sub-study. (In R’s metafor package these would be called “mean differences” and desigated using “MD”).</p>
<p>Another effect size included in the data is the <strong>log-response ratio</strong>. Once your analysis is up and running we’ll discuss this more. (In metafor this is called the “Ratio of mean”, or “ROM”).</p>
<p>Setting up the data as a difference changes the hypothesis testing framework a little bit. Normally when you have an experimental group and a control group you use a 2-sample t-test to compare the groups. The p-value from this test is used to gauge whether the 2 groups are different from each other.</p>
<p>For these data, we are turning data on 2 groups and changing it into a 1 value: the differene between the groups. We therefore will test whether this difference is different from zero using 1-sample t-tests. How these t-tests are exactly formulated depends on the set up of the model, but the math is the same as a 1-sample t-test.</p>
</div>
<div id="variable-sample-sizes-precision-of-effect-sizes" class="section level3">
<h3 class="hasAnchor">
<a href="#variable-sample-sizes-precision-of-effect-sizes" class="anchor"></a>Variable sample sizes &amp; precision of effect sizes</h3>
<p>One additional wrinkle of this data set is that the sample size (N) varies a lot depending on chance and also the allele. For example, several students had less than 5 male experimental flies, while one had 25 male experimental flies.</p>
<p>The difference scores that comprise this data therefore have widely variable sample sizes. Each difference score can be thought of as an estimate of the impact of a mutation in the candidate gene on the wing phenotype. This estimate of the true effect, however, is imprecise due to random variation. Moreover, small sample sizes exacerbate the impacts of sampling variation. So, some of our 252 estimates of the impacts of candidate gene mutations are probably pretty representative of the true impacts of the mutations, while others, especially those with small sample sizes, are probably not very good estimates.</p>
<p>If we were just to analyze the difference scores at face value we would be treating each data point as having the same level of precision and the same degree of bias. For many normal scientific situations this is a reasonable assumption. If we measure the absorbance of 252 samples of protein using the same spectrophotometer, we’d expect each sample to have been measured with the same level of precision and any bias to be the same for all samples. Our wing data aren’t like this because each student could have a different degree of bias and sample sizes vary widely; we should account for this in our analysis.</p>
</div>
<div id="pooled-standard-deviations" class="section level3">
<h3 class="hasAnchor">
<a href="#pooled-standard-deviations" class="anchor"></a>Pooled standard deviations</h3>
<p>The problem of small and unequal sample sizes is a common one, and depending on how the data are formatted and the study goals different approaches can be used. Our difference scores are a type of <strong>effect size</strong> - the simplest kind, just the difference between 2 means. Just as we can take a mean of each student’s data, we can also take the standard deviation (SD) of the data. For our effect size, we subtract control from experimental flies to get single number that characterizes the outcome of the study. For the standard deviation, we get a single value the characterizes the study by calcualting the <strong>pooled standard deviation</strong>. This is a combination of the 2 standard deviations (to be specific, a <a href="https://en.wikipedia.org/wiki/Weighted_arithmetic_mean">weighted mean</a> of the 2 SDs).</p>
<p><a href="https://en.wikipedia.org/wiki/Pooled_variance">Pooled variances</a> and <a href="https://support.minitab.com/en-us/minitab/18/help-and-how-to/statistics/basic-statistics/supporting-topics/data-concepts/what-is-the-pooled-standard-deviation/">pooled standard deviations</a> are common in statistical calculations, but usually we don’t think about them because the math is taken care of by the computer. With effect sizes its important that we think more explicitly about it because each effect size has its own associated poole standard devaition.</p>
</div>
<div id="weighted-regression" class="section level3">
<h3 class="hasAnchor">
<a href="#weighted-regression" class="anchor"></a>Weighted regression</h3>
<p>Once we have a difference score and a pooled standard deviation, we can do something called <strong>weighted regression</strong>. Meta-analysis is a type of weighted regression. Analyses which integrate phylogentic information are another form. Weighted regression is like standard t-tests, ANOVA, and regression, except that it gives data estimated with higher precision (smaller SD) more influence in the analysis. To facilitate analysis I have already calcualted the pooled standard deviaiton for you in some version of the data.</p>
<p>(Another way to do weighted regression is to use the sample size to set the weights. Methods to do this have been established when using the log response ratio (aka ratio of means “ROM”) as an effect size.)</p>
<p>I forget why what follows is in this particular file; I will leave it in until I remember what I want to do wiht it : )</p>
</div>
<div id="student-vs-fly-level-data" class="section level3">
<h3 class="hasAnchor">
<a href="#student-vs-fly-level-data" class="anchor"></a>Student vs. fly-level data</h3>
<p>Because these data result from summarizing the work done by each student, I will refer to it as <strong>student-level data</strong>. In contrast, the raw data generated by the student is <strong>fly-level</strong> data because each data point represents a single fly.</p>
</div>
<div id="key-issues" class="section level3">
<h3 class="hasAnchor">
<a href="#key-issues" class="anchor"></a>Key issues</h3>
<p>The primary question: which alleles of which genes produce the biggest effect? Everyone working with these data should evaluate this. There are a number additinaly issues with how to analyze the data that</p>
<p>Since these data are uniquely structured</p>
<ol style="list-style-type: decimal">
<li>
<strong>Which effect size is best?</strong>: The default approach for the analysis of this data is to take the difference between the mean of each student’s personal dataset and then average accross the all of the students. This <strong>difference between means</strong> or <strong>difference score</strong> is a type of <strong>unstandardized effect size.</strong> It is easy to calcualte and understand. However, using a difference score ignores differences in sample size between each student’s personal dataset. One</li>
<li>What type of <strong>multiple comparisons correction</strong> should be used?</li>
<li><strong>Should data be weighted?</strong></li>
<li><p>Parametic versus non-paranetric analyses</p></li>
<li>Compare the effects of different approaches to multiple comparisons correction: None, Bonferroni, Holm, and the false discovery rate (FDR). You can pick any effect size you want (difference of means, Cohen’s D, log reponse ratio) and any set of weights appropriate for that measure (eg, for difference in means you could use no weights, standard deviations, or sample size).</li>
<li>
</ol>
</div>
</div>
<div id="analysis" class="section level2">
<h2 class="hasAnchor">
<a href="#analysis" class="anchor"></a>Analysis</h2>
<div id="difference-scores-w-pooled-sd-weights" class="section level3">
<h3 class="hasAnchor">
<a href="#difference-scores-w-pooled-sd-weights" class="anchor"></a>Difference scores w/ pooled SD weights</h3>
<p>these notes not complted</p>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m.diffs &lt;-<span class="st"> </span><span class="kw">lm</span>(diff <span class="op">~</span><span class="st"> </span><span class="op">-</span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>gene.allele, 
           <span class="dt">data =</span> data_summ)

m.diffs.wt &lt;-<span class="st"> </span><span class="kw">lm</span>(diff <span class="op">~</span><span class="st"> </span><span class="op">-</span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>gene.allele,
                  <span class="dt">weights =</span> <span class="dv">1</span><span class="op">/</span>sd.pool, 
           <span class="dt">data =</span> data_summ)

<span class="kw">plot</span>(<span class="kw">coef</span>(m.diffs),
<span class="kw">coef</span>(m.diffs.wt))

diffs.mod.tab &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">summary</span>(m.diffs.wt)<span class="op">$</span>coefficients)

<span class="kw">names</span>(diffs.mod.tab) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"estimate"</span>,<span class="st">"SE"</span>,<span class="st">"t"</span>,<span class="st">"P"</span>)</code></pre></div>
</div>
<div id="compare-p-valuve-corrections" class="section level3">
<h3 class="hasAnchor">
<a href="#compare-p-valuve-corrections" class="anchor"></a>Compare p-valuve corrections</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">y &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">gene =</span> <span class="kw">row.names</span>(diffs.mod.tab),
      <span class="dt">bonf.y =</span> <span class="kw">round</span>(<span class="kw">p.adjust</span>(mod.tab<span class="op">$</span>P,
               <span class="dt">method =</span> <span class="st">"bonferroni"</span>),<span class="dv">4</span>)

,<span class="dt">holm.y =</span> <span class="kw">round</span>(<span class="kw">p.adjust</span>(diffs.mod.tab<span class="op">$</span>P,
               <span class="dt">method =</span> <span class="st">"holm"</span>),<span class="dv">4</span>)

,<span class="dt">fdr.y =</span> <span class="kw">round</span>(<span class="kw">p.adjust</span>(diffs.mod.tab<span class="op">$</span>P,
               <span class="dt">method =</span> <span class="st">"fdr"</span>),<span class="dv">4</span>))</code></pre></div>
</div>
<div id="log-response-ratios" class="section level3">
<h3 class="hasAnchor">
<a href="#log-response-ratios" class="anchor"></a>log response Ratios</h3>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">m.log.RR &lt;-<span class="st"> </span><span class="kw">lm</span>(log.RR <span class="op">~</span><span class="st"> </span><span class="op">-</span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>gene.allele, 
           <span class="dt">data =</span> data_summ)

m.log.RR.wt &lt;-<span class="st"> </span><span class="kw">lm</span>(log.RR <span class="op">~</span><span class="st"> </span><span class="op">-</span><span class="dv">1</span> <span class="op">+</span><span class="st"> </span>gene.allele,
                  <span class="dt">weights =</span> logRR.weight, 
           <span class="dt">data =</span> data_summ)

<span class="kw">plot</span>(<span class="kw">coef</span>(m.log.RR),
<span class="kw">coef</span>(m.log.RR.wt))



logRR.mod.tab &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="kw">summary</span>(m.log.RR.wt)<span class="op">$</span>coefficients)

<span class="kw">names</span>(logRR.mod.tab) &lt;-<span class="st"> </span><span class="kw">c</span>(<span class="st">"estimate"</span>,<span class="st">"SE"</span>,<span class="st">"t"</span>,<span class="st">"P"</span>)</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">x &lt;-<span class="st"> </span><span class="kw">data.frame</span>(<span class="dt">gene =</span> <span class="kw">row.names</span>(logRR.mod.tab),
      <span class="dt">bonf =</span> <span class="kw">round</span>(<span class="kw">p.adjust</span>(logRR.mod.tab<span class="op">$</span>P,
               <span class="dt">method =</span> <span class="st">"bonferroni"</span>),<span class="dv">4</span>)

,<span class="dt">holm =</span> <span class="kw">round</span>(<span class="kw">p.adjust</span>(logRR.mod.tab<span class="op">$</span>P,
               <span class="dt">method =</span> <span class="st">"holm"</span>),<span class="dv">4</span>)

,<span class="dt">fdr =</span> <span class="kw">round</span>(<span class="kw">p.adjust</span>(logRR.mod.tab<span class="op">$</span>P,
               <span class="dt">method =</span> <span class="st">"fdr"</span>),<span class="dv">4</span>))</code></pre></div>
<div class="sourceCode"><pre class="sourceCode r"><code class="sourceCode r">
m0 &lt;-<span class="st"> </span><span class="kw">lmer</span>(diff <span class="op">~</span><span class="st"> </span>sex <span class="op">+</span><span class="st"> </span><span class="kw">factor</span>(temp.C)<span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>gene.allele),
           <span class="co">#weights = logRR.weight,</span>
           <span class="dt">data =</span> data_summ)

m1 &lt;-<span class="st"> </span><span class="kw">lmer</span>(diff <span class="op">~</span><span class="st"> </span>sex<span class="op">*</span><span class="kw">factor</span>(temp.C) <span class="op">+</span><span class="st"> </span>(<span class="dv">1</span><span class="op">|</span>gene.allele), 
           <span class="co">#weight = logRR.weight,</span>
           <span class="dt">data =</span> data_summ)

<span class="kw">anova</span>(m0,m1)</code></pre></div>
</div>
</div>
  </div>

  <div class="col-md-3 hidden-xs hidden-sm" id="sidebar">
        <div id="tocnav">
      <h2 class="hasAnchor">
<a href="#tocnav" class="anchor"></a>Contents</h2>
      <ul class="nav nav-pills nav-stacked">
<li><a href="#introduction">Introduction</a></li>
      <li><a href="#analysis">Analysis</a></li>
      </ul>
</div>
      </div>

</div>


      <footer><div class="copyright">
  <p>Developed by First Last.</p>
</div>

<div class="pkgdown">
  <p>Site built with <a href="http://pkgdown.r-lib.org/">pkgdown</a>.</p>
</div>

      </footer>
</div>

  

  
</body>
</html>
